src/main/java/de/tum/in/www1/artemis/service/LearningGoalService.java

```diff
@@ -81,4 +81,113 @@ public SearchResultPageDTO<LearningGoal> getAllOnPageWithSize(final PageableSear
         }
         return new SearchResultPageDTO<>(lecturePage.getContent(), lecturePage.getTotalPages());
     }
+
+    /**
+     * Checks if the provided learning goals and relations between them contain a cycle
+     *
+     * @param learningGoals The set of learning goals that get checked for cycles
+     * @param relations     The set of relations that get checked for cycles
+     * @return A boolean that states whether the provided learning goals and relations contain a cycle
+     */
+    public boolean doesCreateCircularRelation(Set<LearningGoal> learningGoals, Set<LearningGoalRelation> relations) {
+        // Inner class Vertex is only used in this method for cycle detection
+        class Vertex {
```

Comment:
This can be a record

Reply:
If I change it to a record I would not be able to set and unset the beingVisited flag. Or am I missing something?